src/classes/CON_CannotDelete_TDTM.cls

```diff
@@ -13,60 +13,79 @@
+    * @param listOld the list of Contacts from trigger old.
+    * @param triggerAction which trigger event (BeforeInsert, AfterInsert, etc.).
+    * @param objResult the describe for Contact
+    * @return dmlWrapper.
     ********************************************************************************************************/
-    public override DmlWrapper run(List<SObject> newlist, List<SObject> oldlist, 
+    public override DmlWrapper run(List<SObject> newlist, List<SObject> oldlist,
     TDTM_Runnable.Action triggerAction, Schema.DescribeSObjectResult objResult) {
 
-        Map<ID, SObject> oldmap = new Map<ID, SObject>(oldlist);
-        
-        if (triggerAction == TDTM_Runnable.Action.BeforeDelete) {
-                        
-            for (Contact c : [select ID, (select ID from Contact.Affl_Accounts__r LIMIT 1), 
-                                         (select ID from Contact.Program_Enrollments__r LIMIT 1), 
-                                         (select ID from Contact.Student_Course_Enrollments__r LIMIT 1) 
-                                         from Contact where ID in :oldlist]) {
-                
-                if(c.Affl_Accounts__r.size() > 0 || c.Program_Enrollments__r.size() > 0 
-                || c.Student_Course_Enrollments__r.size() > 0) {
+      Map<Id, Contact> oldMap = new Map<Id, Contact>((List<Contact>)oldList);
+      Boolean isConDeletionPreventionEnabled = UTIL_CustomSettingsFacade.getSettings().Prevent_Contact_Deletion__c;
 
-                    Contact contactInContext = (Contact)oldmap.get(c.ID);
+        if (triggerAction == TDTM_Runnable.Action.BeforeDelete && isConDeletionPreventionEnabled) {
+          for (Contact c : [SELECT ID,
+                        (SELECT ID FROM Contact.Addresses__r LIMIT 1),
+                        (SELECT ID FROM Contact.Affl_Accounts__r LIMIT 1),
+                        (SELECT ID FROM Contact.Applications_Applicant__r LIMIT 1),
+                        (SELECT ID FROM Contact.Applications_Preparer__r LIMIT 1),
+                        (SELECT ID FROM Contact.Attendance_Events__r LIMIT 1),
+                        (SELECT ID FROM Contact.Attributes__r LIMIT 1),
+                        (SELECT ID FROM Contact.Behavior_Involvements__r LIMIT 1),
+                        (SELECT ID FROM Contact.Contact_Languages__r LIMIT 1),
+                        (SELECT ID FROM Contact.Courses_Taught__r LIMIT 1),
+                        (SELECT ID FROM Contact.Program_Enrollments__r LIMIT 1),
+                        (SELECT ID FROM Contact.Relationships__r LIMIT 1),
+                        (SELECT ID FROM Contact.Student_Course_Enrollments__r LIMIT 1),
+                        (SELECT ID FROM Contact.Term_Grades__r LIMIT 1),
+                        (SELECT ID FROM Contact.Tests__r LIMIT 1)
+                        FROM Contact
+                        WHERE ID in :oldlist])
+        {
+        	if (c.Addresses__r.size() > 0 || c.Affl_Accounts__r.size() > 0 ||
```

Comment:
I know it seems like it might be an odd ask, but can we move these checks to a method that does the checks and returns either true or false? That just seems like a ton of parameters. Something like:

if(hasChildObjects(con)) { ... }